tmt/steps/discover/fmf.py

```diff
@@ -251,13 +244,15 @@ def go(self) -> None:
         dist_git_merge = self.get('dist-git-merge', False)
         dist_git_init = self.get('dist-git-init', False)
         dist_git_extract = self.get('dist-git-extract', None)
-        dist_git_remove_fmf_root = self.get('dist-git-remove-fmf-root', False)
 
         # Self checks
         if dist_git_source and not dist_git_merge and (ref or url):
             raise tmt.utils.DiscoverError(
                 "Cannot manipulate with dist-git without "
                 "the `--dist-git-merge` option.")
+        # normalize `dist_git_extract` to list of strings
+        if type(dist_git_extract) == str:
+            dist_git_extract = [dist_git_extract]
```

Comment:
I'd recommend using `tmt.utils.normalize_path_list()` (https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L4383C5-L4383C24) right away - I forgot `discover` plugins do now use our `field()` implementation yet, therefore normalization is not yet "out of the box", but this function would be the best to take care of the input. It would give you a list of `Path` instances, no need for checking type of `dist_git_extract` before calling it.

Reply:
Woud it work with glob strings?

The other thing is that I have the accumulator and the glob list merged in the same input `dist-git-extract`. I feel it would be confusing for the user to have it separated as it might indicate exclusivity between the options, different default logics, etc. But in allowing for that, that function cannot be used right?

I think it's not a big deal for it to be actaul paths at that point because it will always go through glob pattern parsing.

Reply:
See https://github.com/teemtee/tmt/pull/2098/files#r1241278339, this will need more work. I missed the added `src/dst` support, normalization will need to be more complex.